Sources/IO/Image/TIFFReader/index.js

```diff
@@ -0,0 +1,155 @@
+  publicAPI.setUrl = (url, option = { binary: true }) => {
+    model.url = url;
+
+    // Remove the file in the URL
+    const path = url.split('/');
+    path.pop();
+    model.baseURL = path.join('/');
+
+    model.compression = option.compression;
+
+    // Fetch metadata
+    return publicAPI.loadData({
+      progressCallback: option.progressCallback,
+    });
+  };
+
+  // Fetch the actual data arrays
+  publicAPI.loadData = (option = {}) => {
+    const promise = fetchData(model.url, option);
+    promise.then(publicAPI.parse);
+    return promise;
+  };
+
+  publicAPI.parse = (content) => {
+    publicAPI.parseAsArrayBuffer(content);
+  };
+
+  publicAPI.parseAsArrayBuffer = (content) => {
+    if (!content) {
+      return;
+    }
+
+    // Read Header
+    const ifds = UTIF.decode(content);
+    UTIF.decodeImage(content, ifds[0]);
+    const data = UTIF.toRGBA8(ifds[0]);
+
+    const width = ifds[0].width;
+    const height = ifds[0].height;
+    const output = new Uint8Array(data.length);
+
+    if (model.flipY) {
+      for (let y = 0; y < height; y++) {
+        for (let x = 0; x < width; x++) {
+          const srcIndex = (y * width + x) * 4;
+          const destIndex = ((height - y - 1) * width + x) * 4;
+
+          output[destIndex] = data[srcIndex]; // R
+          output[destIndex + 1] = data[srcIndex + 1]; // G
+          output[destIndex + 2] = data[srcIndex + 2]; // B
+          output[destIndex + 3] = data[srcIndex + 3]; // A
+        }
+      }
+    }
+
+    const dataExtent = [0, width - 1, 0, height - 1];
+    const dataSpacing = [1, 1, 1];
+
+    const imageData = vtkImageData.newInstance();
+    imageData.setDimensions(width, height, 1);
+    imageData.setExtent(dataExtent);
+    imageData.setSpacing(dataSpacing);
+
+    const dataArray = vtkDataArray.newInstance({
+      name: 'TIFFImage',
+      numberOfComponents: 4,
+      values: output,
+    });
+
+    imageData.getPointData().setScalars(dataArray);
+    model.output[0] = imageData;
+  };
+
+  publicAPI.requestData = (inData, outData) => {
+    publicAPI.parse(model.parseData);
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  flipY: true,
+};
+
+// ----------------------------------------------------------------------------
+
+export function extend(publicAPI, model, initialValues = {}) {
+  Object.assign(model, DEFAULT_VALUES, initialValues);
+
+  // Make this a VTK object
+  macro.obj(publicAPI, model);
+
+  // Also make it an algorithm with one input and one output
+  macro.algo(publicAPI, model, 0, 1);
+
+  macro.get(publicAPI, model, ['url', 'baseURL']);
+  macro.setGet(publicAPI, model, ['dataAccessHelper', 'flipY']);
+
+  // Object specific methods
+  vtkTIFFReader(publicAPI, model);
+
+  // To support destructuring
+  if (!model.compression) {
+    model.compression = null;
```

Comment:
not sure why you need to enforce setting it to null.
Why undefined is not ok?
One way to do it is to specify compression in DEFAULT_VALUES?

Reply:
that's how it's done in other readers, do you want me to change it ? 

https://github.com/Kitware/vtk-js/blob/master/Sources/IO/Geometry/STLReader/index.js#L360-L367
https://github.com/Kitware/vtk-js/blob/master/Sources/IO/Geometry/DracoReader/index.js#L323-L330

Reply:
it seems that it is to allow `const { compression, progressCallback } = model;` in the DracoReader.

I suggest you move them all in DEFAULT_VALUE.